### PR TITLE
Fix EvaluatedNodes in ScheduleResult

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
If some(not all) nodes are marked as unschedulable in filterPlugin, EvaluatedNodes in SchedulerResult is incorrect.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/lgtm
/assign @Huang-wei

Good catch!

File: pkg/scheduler/core/generic_scheduler_test.go [hunk 1]
```diff
@@ -659,6 +662,9 @@ func TestGenericScheduler(t *testing.T) {
 			if test.expectedHosts != nil && !test.expectedHosts.Has(result.SuggestedHost) {
 				t.Errorf("Expected: %s, got: %s", test.expectedHosts, result.SuggestedHost)
 			}
+			if test.wErr == nil && len(test.nodes) != result.EvaluatedNodes {
+				t.Errorf("Expected EvaluatedNodes: %d, got: %d", len(test.nodes), result.EvaluatedNodes)
```
Comment (hex108) [inline pkg/scheduler/core/generic_scheduler_test.go]: Nit: it'll be better to print the test name in the log.

**Review by hex108 — APPROVED** [review]

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

Thanks, @mrkm4ntr .

File: pkg/scheduler/core/generic_scheduler_test.go (hunk 1 above)
Comment (mrkm4ntr) [reply pkg/scheduler/core/generic_scheduler_test.go]: Test name is already printed like below.
```
--- FAIL: TestGenericScheduler (0.00s)
    --- FAIL: TestGenericScheduler/test_with_partial_failed_filter_plugin (0.00s)
        generic_scheduler_test.go:666: Expected EvaluatedNodes: 2, got: 1
```

**Comment by mrkm4ntr** [comment]
@draveness Could you add lgtm label again? I resolved conflicts.

**Review by draveness — APPROVED** [review]
/lgtm 

Thanks

**Comment by liggitt** [comment]
/hold

this is failing batch builds (see https://prow.k8s.io/?repo=kubernetes%2Fkubernetes&type=batch&pull=master%2C82040%2C82053%2C82054%2C82060%2C82064%2C82095%2C82113%2C82121%2C82156%2C82161%2C82170%2C82175%2C82187%2C82193%2C82209)

**Comment by alejandrox1** [comment]
pull-kubernetes-bazel-test failed with the following: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/82156/pull-kubernetes-bazel-test/1171761969833709568#0:build-log.txt%3A57
```
/b/f/w/pkg/scheduler/core/generic_scheduler_test.go:1384:16: filterPlugin.returnCode undefined (type *FakeFilterPlugin has no field or method returnCode)
```

**Comment by mrkm4ntr** [comment]
I'm fixing this.

**Comment by mrkm4ntr** [comment]
@draveness Could you add lgtm label again? I resolved conflict again...

**Review by draveness — COMMENTED** [review]
/lgtm

**Pull request merged**